napari/components/overlays/interaction_box.py

```diff
@@ -1,93 +1,43 @@
-        Holds an affine transform that is modified by interaction with the vertices of the box
-    selection_box_drag : list
-        4x2 list of a box that is being drawn to select items. Gets updated during the mouse-drag
-    selection_box_final : list
-        4x2 list of a box that has been drawn to select item. Gets updated once the mouse drag is finished
-    transform_start : napari.util.transforms.Affine
-        The affine transformation of the box before a new mouse drag was started. Gets updated when a new drag begins.
-    transform_drag : napari.util.transforms.Affine
-        The affine transformation of the box during a mouse drag. Gets updated when mouse moves during drag.
-    transform_final : napari.util.transforms.Affine
-        The affine transformation of the box after a mouse drag. Gets updated when a mouse drag is finished.
-    allow_new_selection: bool
-        Bool indicating whether the interaction box allows creating a new selection by dragging outside an existing interaction_box.
+    visible : bool
+        If the box is visible or not.
+    bounds : 2-tuple of 2-tuples
+        Corners at top left and bottom right in layer coordinates.
+    handles : bool
+        Whether to show the handles for transfomation or just the box.
+    selected_handle : Optional[InteractionBoxHandle]
+        The currently selected handle.
     """
 
-    points: Array[float, (-1, 2)] = None
-    show: bool = False
-    show_handle: bool = False
-    show_vertices: bool = False
-    selection_box_drag: Array[float, (4, 2)] = None
-    selection_box_final: Array[float, (4, 2)] = None
-    transform_start: Affine = Affine()
-    transform_drag: Affine = Affine()
-    transform_final: Affine = Affine()
-    transform: Affine = Affine()
-    allow_new_selection: bool = True
-    selected_vertex: int = None
+    bounds: Tuple[Tuple[float, float], Tuple[float, float]] = ((0, 0), (0, 0))
+    handles: bool = False
+    selected_vertex: Optional[InteractionBoxHandle] = None
 
-    @property
-    def _box(self):
-        box = self._create_box_from_points()
-        if box is None:
-            return None
-        if self.transform:
-            return self.transform(box)
-        return box
+    def update_from_points(self, points):
+        """Create as a bounding box of the given points"""
+        self.bounds = calculate_bounds_from_contained_points(points)
 
-    def _create_box_from_points(self):
-        """Creates the axis aligned interaction box from the list of points"""
-        if self.points is None or len(self.points) < 1:
-            return None
 
-        data = self.points
+class TransformBoxOverlay(SceneOverlay):
+    """A box that can be used to transform layers.
 
-        min_val = np.array([data[:, 0].min(axis=0), data[:, 1].min(axis=0)])
-        max_val = np.array([data[:, 0].max(axis=0), data[:, 1].max(axis=0)])
+    Attributes
+    ----------
+    visible : bool
+        If the box is are visible or not.
```

Comment:
I do not see where `visible` is defined. Also, I do not see `opacity`, `order` `position`, and `selected_vertex` in the docstring.